generator/ServiceClientGeneratorLib/DefaultConfiguration/DefaultConfigurationController.cs

```diff
@@ -52,33 +51,11 @@ public DefaultConfigurationModel LoadDefaultConfiguration(string repositoryRootD
             }
 
             var parsedModel = _defaultConfigurationParser.Parse(json);
-
-            EnrichLegacyMode(parsedModel);
-
+            
+            var validModes = parsedModel.Modes.Where(m => m.Name.ToPascalCase() != "Legacy").ToList();
```

Comment:
Not sure if we should add this part, maybe we will need to add a new mode named `Legacy` in the future for any reason.

Reply:
I think that's unlikely, `Legacy` is already one of the modes (https://docs.aws.amazon.com/sdkref/latest/guide/feature-smart-config-defaults.html), if a new one is added in the future we'd have to pick a new name.

Either way, it'd require us to update `generator/ServiceClientGeneratorLib/DefaultConfiguration/DefaultConfigurationModel.cs` as well.